Have the DrawingAreaProxy control `delegatesScrolling` in the UI process

smfr opened the pull request:
#### 263b327294f3c5e15fe23efd03d499cde6284569
<pre>
Have the DrawingAreaProxy control `delegatesScrolling` in the UI process
<a href="https://bugs.webkit.org/show_bug.cgi?id=246243">https://bugs.webkit.org/show_bug.cgi?id=246243</a>
rdar://100924685

Reviewed by Chris Dumez.

Move the notion of `delegatesScrolling` from WebPageProxy to DrawingAreaProxy, since this is presentation-related.
This was only actually used in RemoteLayerTreeDrawingAreaProxy to control the indicator position, but future patches will make more use of it.

Also have the DrawingAreaProxy be the factory for the RemoteScrollingCoordinatorProxy.

* Source/WebKit/UIProcess/API/mac/WKContentViewMac.mm:
(-[WKContentView initWithFrame:page:]): No need to call setDelegatesScrolling(); this is derived from the DrawingAreaProxy.
* Source/WebKit/UIProcess/DrawingAreaProxy.h:
(WebKit::DrawingAreaProxy::usesDelegatedScrolling const):
* Source/WebKit/UIProcess/RemoteLayerTree/RemoteLayerTreeDrawingAreaProxy.h:
* Source/WebKit/UIProcess/RemoteLayerTree/RemoteLayerTreeDrawingAreaProxy.mm:
(WebKit::RemoteLayerTreeDrawingAreaProxy::createScrollingCoordinatorProxy const):
(WebKit::RemoteLayerTreeDrawingAreaProxy::indicatorLocation const): RemoteLayerTreeDrawingAreaProxy always uses delegated scrolling so remove the condition.
(WebKit::RemoteLayerTreeDrawingAreaProxy::updateDebugIndicator): Ditto
* Source/WebKit/UIProcess/WebPageProxy.cpp:
(WebKit::WebPageProxy::setDrawingArea):
* Source/WebKit/UIProcess/WebPageProxy.h:
* Source/WebKit/UIProcess/ios/WKContentView.mm:
(-[WKContentView _commonInitializationWithProcessPool:configuration:]): No need to call setDelegatesScrolling(); this is derived from the DrawingAreaProxy.

Canonical link: <a href="https://commits.webkit.org/255356@main">https://commits.webkit.org/255356@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/ddaab0ff84f0edc30cd436592e372c1ea0fcb20f

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/92108 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/1339 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/22618 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/101895 "Built successfully") | 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/1337 "Built successfully") | [✅ 🛠 mac-debug](https://ews-build.webkit.org/#/builders/71/builds/29730 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/84546 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/98067 "Built successfully") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/97766 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/78/builds/855 "Passed tests") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/78631 "Built successfully") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/27783 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/9/builds/82769 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/82363 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/34/builds/70827 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/66/builds/36157 "Built successfully") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/73/builds/16378 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/67/builds/33914 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/17480 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/3714 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/63/builds/37784 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/60/builds/40191 "Passed tests") | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/65/builds/39684 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/36606 "Passed tests") | | 
<!--EWS-Status-Bubble-End-->

cdumez commented on Source/WebKit/UIProcess/WebPageProxy.cpp:
```diff
@@ -1150,8 +1150,8 @@ void WebPageProxy::setDrawingArea(std::unique_ptr<DrawingAreaProxy>&& drawingAre
 #if ENABLE(ASYNC_SCROLLING) && PLATFORM(COCOA)
-    if (m_drawingArea->type() == DrawingAreaType::RemoteLayerTree) {
-        m_scrollingCoordinatorProxy = makeUnique<RemoteScrollingCoordinatorProxy>(*this);
+    if (is<RemoteLayerTreeDrawingAreaProxy>(m_drawingArea)) {
```
This is a case where dynamicDowncast results in nicer code I think:
```
if (auto* drawingArea = dynamicDowncast<RemoteLayerTreeDrawingAreaProxy>(m_drawingArea))
    m_scrollingCoordinatorProxy = drawingArea->createScrollingCoordinatorProxy();
```

cdumez reviewed (approved):
LGTM

smfr committed:
Have the DrawingAreaProxy control `delegatesScrolling` in the UI process
https://bugs.webkit.org/show_bug.cgi?id=246243
rdar://100924685

Reviewed by Chris Dumez.

Move the notion of `delegatesScrolling` from WebPageProxy to DrawingAreaProxy, since this is presentation-related.
This was only actually used in RemoteLayerTreeDrawingAreaProxy to control the indicator position, but future patches will make more use of it.

Also have the DrawingAreaProxy be the factory for the RemoteScrollingCoordinatorProxy.

* Source/WebKit/UIProcess/API/mac/WKContentViewMac.mm:
(-[WKContentView initWithFrame:page:]): No need to call setDelegatesScrolling(); this is derived from the DrawingAreaProxy.
* Source/WebKit/UIProcess/DrawingAreaProxy.h:
(WebKit::DrawingAreaProxy::usesDelegatedScrolling const):
* Source/WebKit/UIProcess/RemoteLayerTree/RemoteLayerTreeDrawingAreaProxy.h:
* Source/WebKit/UIProcess/RemoteLayerTree/RemoteLayerTreeDrawingAreaProxy.mm:
(WebKit::RemoteLayerTreeDrawingAreaProxy::createScrollingCoordinatorProxy const):
(WebKit::RemoteLayerTreeDrawingAreaProxy::indicatorLocation const): RemoteLayerTreeDrawingAreaProxy always uses delegated scrolling so remove the condition.
(WebKit::RemoteLayerTreeDrawingAreaProxy::updateDebugIndicator): Ditto
* Source/WebKit/UIProcess/WebPageProxy.cpp:
(WebKit::WebPageProxy::setDrawingArea):
* Source/WebKit/UIProcess/WebPageProxy.h:
* Source/WebKit/UIProcess/ios/WKContentView.mm:
(-[WKContentView _commonInitializationWithProcessPool:configuration:]): No need to call setDelegatesScrolling(); this is derived from the DrawingAreaProxy.

Canonical link: https://commits.webkit.org/255356@main
